### PR TITLE
Issue 60: Clean up new clang-tidy issues and fix doxygen output

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

**Review by riwoh — APPROVED** [review]
umm..  this is a pretty hefty checkin.  it has so many const additions + whitespace changes for basically every file that it's difficult to find the logic changes so it's easy to miss.  the logic changes I did find seemed to be something that lint or something would have flagged.

I believe I looked through all the code, but I'll be honest, very easy if I missed something :)

File: reference/src/client/include/sa_types.h [hunk 1]
```diff
@@ -140,126 +140,186 @@ typedef struct {
...
+    /** ANSI X9.63 Key Derivation Function Algorithm.
+     * See ANSI X9.63 for definition. */
     SA_KDF_ALGORITHM_ANSI_X963,
+    /** CMAC Key Derivation Function Algorithm--a.k.a. the key based key derivation function (KBKDF).
+     * See NIST SP 800-108 for definition. */
     SA_KDF_ALGORITHM_CMAC,
+    /** Netflix Key Derivation Function Algorithm.
+     * See https://github.com/Netflix/msl/wiki/Pre-shared-Keys-or-Model-Group-Keys-Entity-Authentication for
+     * definition .*/
     SA_KDF_ALGORITHM_NETFLIX
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/include/sa_types.h:236]: Added in the other review.

File: reference/src/client/test/sa_key_derive_common_root_key_ladder.cpp [hunk 2]
```diff
@@ -25,14 +25,16 @@
 
 using namespace client_test_helpers;
```
Comment (mhabrat) [inline reference/src/client/test/sa_key_derive_common_root_key_ladder.cpp:27]: Missing second asterisk removal in file header.

File: reference/src/clientimpl/src/sa_crypto_cipher_process.c [hunk 3]
```diff
@@ -72,28 +66,23 @@ sa_status sa_crypto_cipher_process(
                     break;
                 }
 
+                if (out->context.clear.offset > out->context.clear.length) {
```
Comment (mhabrat) [inline reference/src/clientimpl/src/sa_crypto_cipher_process.c:69]: Are these some of the integer overflow tests from the other PR?

File: reference/src/clientimpl/src/sa_crypto_cipher_process_last.c [hunk 4]
```diff
@@ -74,28 +68,22 @@ sa_status sa_crypto_cipher_process_last(
                     break;
                 }
 
+                if (out->context.clear.offset > out->context.clear.length) {
```
Comment (mhabrat) [inline reference/src/clientimpl/src/sa_crypto_cipher_process_last.c:71]: Are these some of the integer overflow tests from the other PR?

File: reference/src/clientimpl/src/sa_process_common_encryption.c [hunk 5]
```diff
@@ -114,28 +102,24 @@ sa_status sa_process_common_encryption(
                     break;
                 }
 
+                if (samples[i].out->context.clear.offset > samples[i].out->context.clear.length) {
```
Comment (mhabrat) [inline reference/src/clientimpl/src/sa_process_common_encryption.c:105]: Are these some of the integer overflow tests from the other PR?

File: reference/src/taimpl/src/internal/soc_key_container.c [hunk 6]
```diff
@@ -876,7 +876,8 @@ sa_status soc_kc_unwrap(
             break;
         }
 
-        if (parse_header(header_fields, header_fields_count, &header) != SA_STATUS_OK) {
+        status = parse_header(header_fields, header_fields_count, &header);
+        if (status != SA_STATUS_OK) {
             ERROR("unpack_soc_kc failed");
```
Comment (mhabrat) [inline reference/src/taimpl/src/internal/soc_key_container.c:881]: parse_header

File: reference/src/taimpl/src/internal/soc_key_container.c [hunk 7]
```diff
@@ -885,17 +886,20 @@ sa_status soc_kc_unwrap(
...
         if (payload_fields == NULL) {
             ERROR("json_value_as_map failed");
+            status = SA_STATUS_INVALID_KEY_FORMAT;
             break;
         }
 
-        if (parse_payload(payload_fields, payload_fields_count, &payload) != SA_STATUS_OK) {
+        status = parse_payload(payload_fields, payload_fields_count, &payload);
+        if (status != SA_STATUS_OK) {
             ERROR("unpack_soc_kc failed");
```
Comment (mhabrat) [inline reference/src/taimpl/src/internal/soc_key_container.c:903]: parse_payload

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/CMakeLists.txt [hunk 8]
```diff
@@ -24,7 +24,7 @@ include(FetchContent)
 FetchContent_Declare(
         googletest
         GIT_REPOSITORY https://github.com/google/googletest.git
-        GIT_TAG release-1.12.1
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/CMakeLists.txt:27]: Done

File: reference/src/client/include/sa_cenc.h [hunk 9]
```diff
@@ -16,6 +16,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file sa_cenc.h
+ *
+ * This file contains the function declaration for processing content encrypted as specified in the document
+ * ISO/IEC 23001-7 Common encryption.
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/include/sa_cenc.h:23]: Done

File: reference/src/client/include/sa_types.h [hunk 10]
```diff
@@ -207,61 +242,86 @@ typedef enum {
...
     SA_KEY_FORMAT_SYMMETRIC_BYTES = 0,
+    /** EC Private Bytes Key Format - PKCS #8 encoded */
     SA_KEY_FORMAT_EC_PRIVATE_BYTES,
+    /** RSA Private Key Info Format - PKCS #8 encoded */
     SA_KEY_FORMAT_RSA_PRIVATE_KEY_INFO,
+    /** Exported Key Format - encoded in a SOC specific way */
     SA_KEY_FORMAT_EXPORTED,
+    /** SOC Key Format - encoded according to the SOC Specific Key Specification */
     SA_KEY_FORMAT_SOC,
+    /** TypeJ Key Format - encoded according to the SecApi Key Specification */
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/include/sa_types.h:268]: Done

File: reference/src/client/test/client_test_helpers.cpp [hunk 11]
```diff
@@ -1567,482 +1563,488 @@ namespace client_test_helpers {
...
+        bool key_check_hmac(
+                sa_key key,
+                const std::vector<uint8_t>& clear_key) {
+            auto input = random(AES_BLOCK_SIZE);
 
-        return true;
-    }
+            std::vector<uint8_t> sa3hmac(clear_key.size());
+            if (!hmac_sa(sa3hmac, key, input, SA_DIGEST_ALGORITHM_SHA256)) {
+                ERROR("hmac_sha256_sa failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/client_test_helpers.cpp:1681]: Done

File: reference/src/client/test/client_test_helpers.cpp [hunk 12]
```diff
@@ -1567,482 +1563,488 @@ namespace client_test_helpers {
...
+                        &parameters) != SA_STATUS_OK) {
+                ERROR("sa_crypto_cipher_init failed");
+                return false;
+            }
 
-        out.resize(length);
+            size_t length = in.size();
+            sa_buffer in_buffer = {SA_BUFFER_TYPE_CLEAR, {.clear = {in.data(), in.size(), 0}}};
+            if (sa_crypto_cipher_process(nullptr, *cipher, &in_buffer, &length) != SA_STATUS_OK) {
+                ERROR("sa_crypto_cipher_init failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/client_test_helpers.cpp:1839]: Done

File: reference/src/client/test/client_test_helpers.cpp [hunk 13]
```diff
@@ -1567,482 +1563,488 @@ namespace client_test_helpers {
...
 
-        if (clear != out) {
-            ERROR("RSA decrypt failed");
-            return false;
-        }
+            auto out = std::vector<uint8_t>(length);
+            sa_buffer out_buffer = {SA_BUFFER_TYPE_CLEAR, {.clear = {out.data(), out.size(), 0}}};
+            length = in.size();
+            if (sa_crypto_cipher_process(&out_buffer, *cipher, &in_buffer, &length) != SA_STATUS_OK) {
+                ERROR("sa_crypto_cipher_init failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/client_test_helpers.cpp:1847]: Done

File: reference/src/client/test/sa_key_derive_common_root_key_ladder.cpp (hunk 2 above)
Comment (ericeberry) [reply reference/src/client/test/sa_key_derive_common_root_key_ladder.cpp:27]: Done

Comment (ericeberry) [reply reference/src/client/test/sa_key_derive_common_root_key_ladder.cpp:27]: Done

File: reference/src/clientimpl/src/sa_crypto_cipher_process_last.c (hunk 4 above)
Comment (ericeberry) [reply reference/src/clientimpl/src/sa_crypto_cipher_process_last.c:71]: Yes. Removed from here.

File: reference/src/clientimpl/src/sa_process_common_encryption.c (hunk 5 above)
Comment (ericeberry) [reply reference/src/clientimpl/src/sa_process_common_encryption.c:105]: Yes. Removed

File: reference/src/taimpl/src/internal/soc_key_container.c (hunk 6 above)
Comment (ericeberry) [reply reference/src/taimpl/src/internal/soc_key_container.c:881]: Done

File: reference/src/taimpl/src/internal/soc_key_container.c (hunk 7 above)
Comment (ericeberry) [reply reference/src/taimpl/src/internal/soc_key_container.c:903]: Done

File: reference/src/client/include/sa_types.h [hunk 14]
```diff
@@ -140,64 +140,99 @@ typedef struct {
...
     SA_KDF_ALGORITHM_CONCAT,
+    /** ANSI X9.63 Key Derivation Function Algorithm.
+     * See ANSI X9.63 for definition. */
     SA_KDF_ALGORITHM_ANSI_X963,
+    /** CMAC Key Derivation Function Algorithm--a.k.a. the key based key derivation function (KBKDF).
+     * See NIST SP 800-108 for definition. */
     SA_KDF_ALGORITHM_CMAC,
+    /** Netflix Key Derivation Function Algorithm.
+     * See https://github.com/Netflix/msl/wiki/Pre-shared-Keys-or-Model-Group-Keys-Entity-Authentication for
+     * definition .*/
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/include/sa_types.h]: Done

**Review by mhabrat — APPROVED** [review]

**Pull request merged**